### PR TITLE
Add Timesketch event collector module and recipe 

### LINKED ISSUE #929: FR: Add Timesketch Event collector module
Add a collector module that retrieves events from a Timesketch instance.

The module should accept a user query (including datetime range/indices/tags) in SetUp and be able to store the events as either DataFrame or File (CSV) containers.

### PR DESCRIPTION
**Pull request by sydp**
Fixes #929

### CODE REVIEW

File: dftimewolf/lib/collectors/timesketch.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+      token_password: optional password used to decrypt the
+          Timesketch credential storage. Defaults to an empty string since
+          the upstream library expects a string value. An empty string means
+          a password will be generated by the upstream library.
+      endpoint: Timesketch server URL (e.g. http://localhost:5000/).
+          Optional when token_password is provided.
+      username: Timesketch username. Optional when token_password is provided.
+      password: Timesketch password. Optional when token_password is provided.
+    """
+    if not sketch_id:
```
Comment (tomchop) [inline dftimewolf/lib/collectors/timesketch.py:107]: Can we attempt to get this one from a container too?

File: dftimewolf/lib/collectors/timesketch.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      self.ModuleError('Sketch ID is not set.', critical=True)
+
+    if not start_datetime or not end_datetime:
+      self.ModuleError(
+          'Both the start and end datetime must be set.', critical=True)
+
+    if output_format not in _VALID_OUTPUT_FORMATS:
+      self.ModuleError('Output formats not valid.', critical=True)
```
Comment (tomchop) [inline dftimewolf/lib/collectors/timesketch.py:115]: Can we specify which output formats are valid in the error message?

**Review by tomchop — CHANGES_REQUESTED** [review]

File: dftimewolf/lib/collectors/timesketch.py (hunk 1 above)
Comment (sydp) [reply dftimewolf/lib/collectors/timesketch.py:108]: Done

File: dftimewolf/lib/collectors/timesketch.py (hunk 2 above)
Comment (sydp) [reply dftimewolf/lib/collectors/timesketch.py]: Done

**Review by tomchop — APPROVED** [review]

**Pull request merged**